### PR TITLE
Use .carina on all platforms

### LINKED ISSUE #80: Use .carina directory on Windows too
Microsoft has officially blessed using dot folders with their own tools (e.g. dnvm, .net version manager) so we could use it too. 

The only hitch is that if there are windows peeps out there who have already used Carina and downloaded credentials will end up with 2 folders. Since I think the only windows users at the moment are me and don, that shouldn't be much of a problem, just something to document in the release notes.

### PR DESCRIPTION
**Pull request by carolynvs**
Closes #80 

/cc @rgbkrk

### CODE REVIEW

File: credentials.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		return os.Getenv(CarinaHomeDirEnvVar), nil
+	}
+	if os.Getenv(CredentialsBaseDirEnvVar) != "" {
+		return os.Getenv(CredentialsBaseDirEnvVar), nil
+	}
+
+	// Support XDG
+	if os.Getenv(xdgDataHomeEnvVar) != "" {
+		return filepath.Join(os.Getenv(xdgDataHomeEnvVar), defaultNonDotDir), nil
+	}
```
Comment (rgbkrk) [inline credentials.go:24]: Hey as long as we're still supporting an XDG env var to make those folks happy, I definitely prefer this singular home directory setup.

**Comment by rgbkrk** [comment]
The only thing that worries me about our approach, even now, is that we should be writing to `~/Library/carina` since we're storing temp tokens (people back up the dotfiles in their home directory willy nill).

**Comment by carolynvs** [comment]
At the moment, the .carina directory is pretty much a cache though right? If it isn't backed up, the next time they run `carina env goodtimes`, it is just recreated.

**Comment by rgbkrk** [comment]
Yeah, I mostly worry about all the home directories I've found on GitHub where people uploaded Rackspace API keys, Amazon keys, etc. on "accident".

**Pull request merged**